kitsune/sumo/templatetags/jinja_helpers.py

```diff
@@ -507,3 +507,22 @@ def image_for_product(product_slug):
     except Product.DoesNotExist:
         return os.path.join(settings.STATIC_URL, 'products', 'img', 'product_placeholder.png')
     return obj.image_url
+
+
+@jinja2.contextfunction
+@library.global_function
+def show_fx_download(context):
+    """
+    Decides whether or not to render the Firefox download button based on the current
+    product being displayed.
+
+    This function is used in conjunction with show-fx-download.js. This function simply adds
+    markup to the template wrapped in a 'hidden' class. The JS removes the 'hidden' class if
+    the visitor is *NOT* already using Firefox.
+    """
+    product = context.get('product', None)
+
+    if product:
+        return product.slug != 'firefox'
```

Comment:
Isn't this backward?

Reply:
Sigh, no, but it is very confusing. tl;dr - I think I'll rename this method to `show_header_fx_download`.

The download button in the header should always be shown _unless_ the visitor is on a Firefox product page (or any derivative thereof - e.g. a KB article). If the visitor lands on a page where `context['product'].slug == 'Firefox'`, the big, in-page download button will be shown - hence hiding the header download button with this function.

Reply:
Ah! Makes sense now. It's supposed to be "backward" from what I was thinking :)